### PR TITLE
Add impossible IDO pools

### PR DESCRIPTION
**Pull request by 0xKARTOD**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @0xKARTOD 

This PR adds a static view of `impossible_finance_IDO_pools` on bnb chain.

(Also please have a look to this PR https://github.com/duneanalytics/spellbook/pull/2593 if you have time)

File: dbt_project.yml [hunk 1]
```diff
@@ -801,6 +801,9 @@ models:
       polygon:
         +schema: rocifi_polygon
 
+    impossible_finance:
+      +schema: impossible_finance
```
Comment (jeff-dude) [inline dbt_project.yml:805]: please add subdirectory for bnb here, as that is where model lives & add schema for `impossible_finance_bnb`

File: models/impossible_finance/bnb/impossible_finance_IDO_pools.sql [hunk 2]
```diff
@@ -0,0 +1,104 @@
+{{
+    config(
+        alias='IDO_pools',
```
Comment (jeff-dude) [inline models/impossible_finance/bnb/impossible_finance_IDO_pools.sql:3]: please use lowercase on `IDO` for consistency across spellbook. this also applies to the file name itself.

File: models/impossible_finance/bnb/impossible_finance_IDO_pools.sql [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        tags=['static'],
+        post_hook='{{ expose_spells(\'["bnb"]\',
+                                    "project",
+                                    "impossible_finance",
+                                    \'["kartod"]\') }}'
+    )
+}}
+
+SELECT blockchain,
+       lower(address) AS pool_address,
```
Comment (jeff-dude) [inline models/impossible_finance/bnb/impossible_finance_IDO_pools.sql:13]: to be safe, something i've tried to push for new static models, can we add `trim()` function here too?

Comment (jeff-dude) [reply models/impossible_finance/bnb/impossible_finance_IDO_pools.sql:13]: `trim()` could be used on most string columns

File: models/impossible_finance/bnb/impossible_finance_IDO_pools.sql [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+       purchase_date,
+       start_staking_period,
+       end_staking_period,
+       CASE
+            WHEN sale_type IN ('Unlimited IDIA sale')
+            THEN lower('0x0b15Ddf19D47E6a86A56148fb4aFFFc6929BcB89')
+            ELSE lower('0xe9e7CEA3DedcA5984780Bafc599bD69ADd087D56')
+        END           AS accepted_currency
+FROM (VALUES
+    (array('bnb'),'0x69b9737aff26bc528ffee4cc3257407be6a13252', 'Openswap', 'Openswap', 'Standard Sale', 1, timestamp('2021-08-30'), timestamp('2021-08-26'), timestamp('2021-08-26')),
```
Comment (jeff-dude) [inline models/impossible_finance/bnb/impossible_finance_IDO_pools.sql:27]: we've moved away from forcing `array()` usage on `blockchain` column -- please do remove

File: models/impossible_finance/bnb/impossible_finance_bnb_schema.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+version: 2
+
+models:
+  - name: impossible_finance_IDO_pools
```
Comment (jeff-dude) [inline models/impossible_finance/bnb/impossible_finance_bnb_schema.yml:4]: lowercase

File: models/impossible_finance/bnb/impossible_finance_bnb_schema.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      contibutors: kartod
+    config:
+      tags: ['impossible', 'bnb']
+    description: "Impossible finance pools where users send BUSD to buy available allocation"
+    columns:
+      - &blockchain
+        name: blockchain
+        description: "Blockchain"
+
+      - &pool_address
```
Comment (jeff-dude) [inline models/impossible_finance/bnb/impossible_finance_bnb_schema.yml:17]: can we apply a unique test on just address?
if not only address, then use `unique_combo_of_columns` test

**Review by jeff-dude — CHANGES_REQUESTED** [review]
thank you @0xKARTOD for the PR. please see below proposed changes prior to merge

**Review by jeff-dude — CHANGES_REQUESTED** [review]
i've fixed the test to run & now there appears to be duplicates. please determine if address alone is valid for unique or there is truly a duplicate.

File: models/impossible_finance/bnb/impossible_finance_bnb_ido_pools.sql [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    ('bnb','0x98fF066BB5041DF3e8299Ce2eAf44d5a9E4f32E4', 'Ouro', 'Ouro', 'Unlimited BUSD Sale', 8, timestamp('2022-01-03'), timestamp('2021-11-16'), timestamp('2021-12-30')),
+    ('bnb','0x0c8dF3f968eC9F2bf182C41C9D42d79dF4a31857', 'Ouro', 'Ouro', 'Unlimited IDIA Sale', 8, timestamp('2022-01-03'), timestamp('2021-11-16'), timestamp('2021-12-30')),
+    ('bnb','0xc5A2fBFAFA03Bb90eFDE3b44Ce13DD99489CDf91', 'Ouro', 'Ouro', 'Private 100', 8, timestamp('2022-01-03'), timestamp('2021-11-16'), timestamp('2021-12-30')),
+    ('bnb','0x347cB9f91d89d5F3e19f93081a0F4c95F529E853', 'Ouro', 'Ouro', 'Private 200', 8, timestamp('2022-01-03'), timestamp('2021-11-16'), timestamp('2021-12-30')),
+
+    ('bnb','0x53E36E2565e8113dF6c1C675B3bC7eC1788cD1C9', 'IDIA', 'IDIA', 'Unlimited Sale', 9, timestamp('2022-01-10'), timestamp('2021-12-23'), timestamp('2022-02-19')),
+    ('bnb','0x98fF066BB5041DF3e8299Ce2eAf44d5a9E4f32E4', 'IDIA', 'IDIA', 'SDO Standard Sale', 9, timestamp('2022-01-10'), timestamp('2021-12-23'), timestamp('2022-02-19')),
```
Comment (Hosuke) [inline models/impossible_finance/bnb/impossible_finance_bnb_ido_pools.sql:64]: @0xKARTOD Line #58 and Line #64 are having the same address. Please check.

**Comment by jeff-dude** [comment]
closing due to inactivity

**Pull request closed without merging**